### PR TITLE
Electron Application

### PR DESCRIPTION
**Pull request by alexDrinkwater**
Can set settings from a pop up window. Not sure how we can build and distribute it yet.

### CODE REVIEW

**Comment by alexDrinkwater** [comment]
fixes #7

**Comment by alexDrinkwater** [comment]
This is what the settings window looks like: 
image

**Comment by alexDrinkwater** [comment]
Here is an app build from this PR, not sure why its so large at this stage... https://easyupload.io/fngnni

**Comment by jvolker** [comment]
Wow, thank you so much for your work. This looks great! 

I just tested the prefilling part on a Mac and it works really well. I haven't tested to get to the beeping part though. Could anyone please confirm it's still working?

I will check the code a bit more thoroughly tonight when I will hopefully have more time and will try to merge it asap.

Again, thanks so much already!

**Comment by jvolker** [comment]
>  Not sure how we can build and distribute it yet.

I have never used it, but we could use Github's release section to host the builds. But yeah, manually building could become a bit time consuming if there are frequent changes.  I would be happy to do the Mac builds every now and then, though.

Do you know if it's possible to build on Github or elsewhere somehow automatically?

**Comment by alexDrinkwater** [comment]
I think I need to add a way that doesnt break existing users that run in node. It may just be a matter of updating the readme and add a script in the package.json

**Comment by alexDrinkwater** [comment]
@jvolker I just restored the ability to run `npm start` to just run using node without having to build the electron app. I think this will mean we can merge this in and get other Pull Requests merged while we figure out how to package and distribute the build.

File: MIQ-Booking-Assistance/package.json [hunk 1]
```diff
@@ -1,17 +1,56 @@
...
-    "start": "node index.js",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "start": "npx run-func miq-assistance.js start",
+    "start-electron": "electron-forge start",
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "package": "electron-forge package",
+    "make": "electron-forge make"
   },
-  "author": "",
+  "author": "github contributors",
```
Comment (alexDrinkwater) [inline MIQ-Booking-Assistance/package.json:13]: Electron forge gave an error when this was empty

File: MIQ-Booking-Assistance/package.json [hunk 2]
```diff
@@ -1,17 +1,56 @@
...
   "name": "miq-nz-booking-tools",
   "version": "1.0.0",
-  "description": "",
-  "main": "index.js",
+  "description": "A tool to help you book MIQ",
+  "main": "main.js",
   "scripts": {
-    "start": "node index.js",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "start": "npx run-func miq-assistance.js start",
```
Comment (alexDrinkwater) [inline MIQ-Booking-Assistance/package.json:7]: This line allows the continued usage of `npm start`

File: MIQ-Booking-Assistance/preload.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+window.addEventListener('DOMContentLoaded', () => {
```
Comment (jvolker) [inline MIQ-Booking-Assistance/preload.js]: I'm having trouble understanding what the code in this file does. Would be so kind to explain it to me? :) Maybe even put a short comment in there if you don't mind. Thanks.

File: README.md [hunk 4]
```diff
@@ -61,7 +61,30 @@ Register a booking using your regular browser (Firefox, Chrome, etc.) before sta
...
 
-*These defaults can be changed at the top of the file `index.js` under `MIQ-Booking-Assistance` folder. Customization is limited to those options at this stage but can be achieved by extending the script.*
+*These defaults can be changed at the top of the file `miq-assistance.js` under `MIQ-Booking-Assistance` folder. Customization is limited to those options at this stage but can be achieved by extending the script.*
+
+## 3. MIQ Booking Assistance (Electron application, currently in progress)
+
+**Features**  
+This is an electron application that runs the same script above to check for MIQ spots.
+
+*This is more unfinished and probably requires some coding skills.*
```
Comment (jvolker) [inline README.md]: I think we should change that to "This is the recommended way for most people and doesn't require any coding skills" or so.

File: README.md [hunk 5]
```diff
@@ -61,7 +61,30 @@ Register a booking using your regular browser (Firefox, Chrome, etc.) before sta
 7. Quickly book your spot (select the available date, tick the reCAPTCHA and click ‘Next’ to secure your spot) on the same page. Alternatively, refresh the page to turn off the alarm. 
 8. You can close the script/window by pressing CTRL + C on the command line.
 
-*These defaults can be changed at the top of the file `index.js` under `MIQ-Booking-Assistance` folder. Customization is limited to those options at this stage but can be achieved by extending the script.*
+*These defaults can be changed at the top of the file `miq-assistance.js` under `MIQ-Booking-Assistance` folder. Customization is limited to those options at this stage but can be achieved by extending the script.*
+
+## 3. MIQ Booking Assistance (Electron application, currently in progress)
+
+**Features**  
+This is an electron application that runs the same script above to check for MIQ spots.
```
Comment (jvolker) [inline README.md:69]: Once the builds are ready we should add here something along "The apps are available for Windows and macOS: _links to latest releases_"

File: MIQ-Booking-Assistance/miq-assistance.js [hunk 6]
```diff
@@ -110,14 +132,33 @@ async function prepareAndCheckPage(page) {
...
 
     if (found) {
-        console.log("AVAILABLE! Found at: " + new Date().toLocaleString())
+        const status = "AVAILABLE! Found at: " + new Date().toLocaleString();
+        console.log(status)
+        updateElectronStatus('status-count', status);
     } else {
         // reload if nothing was available
-        process.stdout.moveCursor(0, -1) // up one line
-        process.stdout.clearLine(1) // from cursor to end
```
Comment (jvolker) [inline MIQ-Booking-Assistance/miq-assistance.js:117]: Do you think there might be a way to keep using this instead if the script is run without Electron? Not a big deal, if it stays like that though.

**Comment by jvolker** [comment]
> @jvolker I just restored the ability to run `npm start` to just run using node without having to build the electron app. I think this will mean we can merge this in and get other Pull Requests merged while we figure out how to package and distribute the build.

This is a great idea. Let's do this.

The code looks really good to me. Just want to test it one more time tomorrow morning as I need to stop working on this for today. Is it good to merge from your end?

I love your work! Thanks again.

File: MIQ-Booking-Assistance/preload.js (hunk 3 above)
Comment (alexDrinkwater) [reply MIQ-Booking-Assistance/preload.js]: I don't actually think im using this file anymore, I will delete.

File: MIQ-Booking-Assistance/miq-assistance.js (hunk 6 above)
Comment (alexDrinkwater) [reply MIQ-Booking-Assistance/miq-assistance.js:117]: I'll try

File: README.md (hunk 5 above)
Comment (alexDrinkwater) [reply README.md:69]: I agree.

**Comment by alexDrinkwater** [comment]
I have addressed your feedback. I think its ready to merge.

**Pull request merged**